src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/UriExtensions.cs

```diff
@@ -22,6 +23,27 @@ public static string GetAbsoluteOrUNCPath(this Uri uri)
         }
 
         // Absolute paths are usually encoded.
-        return uri.AbsolutePath.Contains("%") ? WebUtility.UrlDecode(uri.AbsolutePath) : uri.AbsolutePath;
+        var absolutePath = uri.AbsolutePath.Contains("%") ? WebUtility.UrlDecode(uri.AbsolutePath) : uri.AbsolutePath;
+
+        if (uri.Scheme.ToLower() == "git")
```

Comment:
Please don't use `ToLower()` to perform case-insensitive string comparson. That will allocate a new string.

```suggestion
        if (string.Equals(uri.Scheme, "git", StringComparison.OrdinalIgnoreCase))
```